src/compiler-node.js

```diff
@@ -1,111 +1,61 @@
-
-var fs = require('fs');
-var walk = require('walk');
+var Bluebird = require('bluebird');
+var fs = Bluebird.promisifyAll(require('fs'));
 var path = require('path');
 var mkdir = require('mkdirp');
 var compiler = require('./compiler');
+var utils = require('./utils');
+mkdirp = Bluebird.promisify(mkdir);
```

Comment:
Missing a `var`? Could just replace it with `Bluebird.promisify(require('mkdirp'))` too I guess.